packages/web3/src/signer/signer.ts

```diff
@@ -118,7 +118,7 @@ export abstract class SignerProviderSimple extends SignerProvider {
     return signResult
   }
 
-  protected abstract getPublicKey(address: string): Promise<string>
+  abstract getPublicKey(address: string): Promise<string>
```

Comment:
Please let me know if the `SignerProviderSimple` can have this method public. Not important much, it would be easier for having a single method for build+sign+submit with just a `PrivateKeyWallet`